bitlum.go

```diff
@@ -0,0 +1,763 @@
+package client
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/shopspring/decimal"
+)
+
+const exchangeID = "bitlum"
+
+// Client is the http://exchange.bitlum.io exchange
+type Client struct {
+	backend
+}
+
+// NewExchangeClient create new bitlum exchange with specified URL and
+// authorization token.
+func NewClient(url string, authToken string) *Client {
+	return &Client{
+		backend: &graphqlBackend{
+			url:       url,
+			authToken: authToken,
+		},
+	}
+}
+
+// ID returns exchange ID
+func (c *Client) ID() string {
+	return exchangeID
+}
+
+// Markets return markets supported by exchange
+func (c *Client) Markets() []string {
+	return []string{
+		"BTCETH",
+		"BTCBCH",
+		"BTCDASH",
+		"BTCLTC",
+	}
+}
+
+// UserID returns exchange user ID, user on behalf which all
+// exchange operations are performing.
+func (c *Client) UserID() (string, error) {
+
+	var req request
+
+	req.Query = `
+		query Me {
+			me {
+			  id
+			}
+		}
+	`
+
+	resp := struct {
+		responseBase
+		Data struct {
+			User struct {
+				ID string
+			} `json:"me"`
+		}
+	}{}
+
+	respJSON, err := c.do(req)
+	if err != nil {
+		return "", errors.New("failed to do request: " + err.Error())
+	}
+
+	if err := json.Unmarshal(respJSON, &resp); err != nil {
+		return "", errors.New("failed to json.Unmarshal resp: " +
+			err.Error())
+	}
+
+	if err := resp.Error(); err != nil {
+		return "", errors.New("bitlum server-side error: " +
+			err.Error())
+	}
+
+	return resp.Data.User.ID, nil
+}
+
+type Ticker struct {
```

Comment:
What is ticker?

Reply:
Its from https://en.wikipedia.org/wiki/Ticker_tape canonical name of stock exchange prices.

Reply:
yeah, I meat lets add comment